spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerAnnotationBeanPostProcessor.java

```diff
@@ -85,48 +109,148 @@ public Object postProcessAfterInitialization(final Object bean, String beanName)
 			public void doWith(final Method method) throws IllegalArgumentException, IllegalAccessException {
 				StreamListener streamListener = AnnotationUtils.findAnnotation(method, StreamListener.class);
 				if (streamListener != null) {
-					Method targetMethod = checkProxy(method, bean);
-					Assert.hasText(streamListener.value(), "The binding name cannot be null");
-					final InvocableHandlerMethod invocableHandlerMethod = messageHandlerMethodFactory.createInvocableHandlerMethod(bean, targetMethod);
-					if (!StringUtils.hasText(streamListener.value())) {
-						throw new BeanInitializationException("A bound component name must be specified");
-					}
-					if (mappedBindings.containsKey(streamListener.value())) {
-						throw new BeanInitializationException("Duplicate @" + StreamListener.class.getSimpleName() +
-								" mapping for '" + streamListener.value() + "' on " + invocableHandlerMethod.getShortLogMessage() +
-								" already existing for " + mappedBindings.get(streamListener.value()).getShortLogMessage());
+					Class<?>[] parameterTypes = method.getParameterTypes();
+					boolean hasInputOrOutputAnnotatedParameters = false;
+					for (int parameterIndex = 0; parameterIndex < parameterTypes.length; parameterIndex++) {
+						MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method,
+								parameterIndex);
+						boolean currentParameterIsAnnotated = methodParameter.getParameterAnnotation(
+								Input.class) != null || methodParameter.getParameterAnnotation(Output.class) != null;
+						if (hasInputOrOutputAnnotatedParameters) {
```

Comment:
I don't believe this would catch the case where the 1st param is *not* annotated even if all subsequent params are... instead maybe it should just iterate a counter and at the end of the loop assert that the count is either 0 or equal to the `parameterTypes.length`